dbms/src/Common/AlignedBuffer.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright 2024 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/AlignedBuffer.h>
+#include <Common/Exception.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int CANNOT_ALLOCATE_MEMORY;
+}
+
+void AlignedBuffer::alloc(size_t size, size_t alignment)
+{
+    void * new_buf;
```

Comment:
> If `alloc()` is called two times without calling `daealloc()` or `reset()`, there will be memory leak. Maybe we can add some check to avoid it?

It's impossible to call `alloc()` two times. Because it's a private function and only be used in constructor and `reset()`.